File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/combat/CmdDuel.java
package com.projectswg.holocore.resources.support.global.commands.callbacks.combat;

import com.projectswg.holocore.intents.gameplay.combat.duel.DuelPlayerIntent;
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback;
import com.projectswg.holocore.resources.support.global.player.Player;
import com.projectswg.holocore.resources.support.objects.swg.SWGObject;
import com.projectswg.holocore.resources.support.objects.swg.creature.CreatureObject;
import org.jetbrains.annotations.NotNull;

public final class CmdDuel implements ICmdCallback {
	
	@Override
	public void execute(@NotNull Player player, SWGObject target, @NotNull String args) {
		if (target == null || target.getOwner() == null)
			return;
		
		CreatureObject creoTarget = (CreatureObject) target;
		
		if (creoTarget.hasSentDuelRequestToPlayer(player.getCreatureObject())) {
			new DuelPlayerIntent(player.getCreatureObject(), creoTarget, DuelPlayerIntent.DuelEventType.ACCEPT).broadcast();
		} else {
			new DuelPlayerIntent(player.getCreatureObject(), creoTarget, DuelPlayerIntent.DuelEventType.REQUEST).broadcast();
		}
	}
	
}

File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/combat/CmdDuel.kt
package com.projectswg.holocore.resources.support.global.commands.callbacks.combat

import com.projectswg.holocore.intents.gameplay.combat.DuelPlayerIntent
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback
import com.projectswg.holocore.resources.support.global.player.Player
import com.projectswg.holocore.resources.support.objects.swg.SWGObject
import com.projectswg.holocore.resources.support.objects.swg.creature.CreatureObject

class CmdDuel : ICmdCallback {
	override fun execute(player: Player, target: SWGObject?, args: String) {
		if (target == null || target.owner == null) return

		val creoTarget = target as CreatureObject

		if (creoTarget.hasSentDuelRequestToPlayer(player.creatureObject)) {
			DuelPlayerIntent(player.creatureObject, creoTarget, DuelPlayerIntent.DuelEventType.ACCEPT).broadcast()
		} else {
			DuelPlayerIntent(player.creatureObject, creoTarget, DuelPlayerIntent.DuelEventType.REQUEST).broadcast()
		}
	}
}
